File: Example/TableViewFactory/CoreData/Bank.m
//
//  Bank.m
//  DTTableViewManager
//
//  Created by Denys Telezhkin on 07.12.13.
//  Copyright (c) 2013 Denys Telezhkin. All rights reserved.
//

#import "Bank.h"

@implementation Bank

@dynamic name;
@dynamic city;
@dynamic zip;
@dynamic state;

+(instancetype)insertBankWithInfo:(NSDictionary *)info
           inManagedObjectContext:(NSManagedObjectContext *)context
{
    Bank * newBank = [NSEntityDescription insertNewObjectForEntityForName:NSStringFromClass([self class])
                                                   inManagedObjectContext:context];
    [newBank fillBankWithInfo:info];
    return newBank;
}

-(void)fillBankWithInfo:(NSDictionary *)info
{
    self.name = info[@"name"];
    self.city = info[@"city"];
    self.zip = info[@"zip"];
    self.state = info[@"state"];
}

@end

File: Example/CoreData/Bank.swift
//
//  Bank.swift
//  DTTableViewManager
//
//  Created by Denys Telezhkin on 20.08.15.
//  Copyright (c) 2015 Denys Telezhkin. All rights reserved.
//

import Foundation
import CoreData

@objc(Bank)
class Bank : NSManagedObject
{
    @NSManaged var name : String
    @NSManaged var city : String
    @NSManaged var zip : Int
    @NSManaged var state : String
    
    convenience init(info : [String:Any], inContext context: NSManagedObjectContext)
    {
        let entity = NSEntityDescription.entity(forEntityName: "Bank", in: context)
        self.init(entity: entity!, insertInto: context)
        name = info["name"] as? String ?? ""
        city = info["city"] as? String ?? ""
        zip = info["zip"] as? Int ?? 0
        state = info["state"] as? String ?? ""
    }
}
